### PR TITLE
scala 2.13 upgrade

### PR DESCRIPTION
**Pull request by jackdingilian**

### CODE REVIEW

File: ratatool-sampling/src/main/scala/com/spotify/ratatool/io/BigQueryIO.scala [hunk 1]
```diff
@@ -84,15 +84,15 @@ object BigQueryIO {
 
 }
 
-private class TableRowIterator(private val iterator: PatchedBigQueryTableRowIterator)
+private class TableRowIterator(private val iter: PatchedBigQueryTableRowIterator)
```
Comment (catherinejelder) [inline ratatool-sampling/src/main/scala/com/spotify/ratatool/io/BigQueryIO.scala:87]: curious why the rename?

Comment (jackdingilian) [reply ratatool-sampling/src/main/scala/com/spotify/ratatool/io/BigQueryIO.scala:87]: yeah this is a bit weird.  2.13 collections update added a final field called iterator to Iterators: https://github.com/scala/scala/blob/v2.13.2/src/library/scala/collection/Iterator.scala#L96

So this is now trying to override that and can't.  This maintains compatibility for both 2.12 and 2.13

File: ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/BigDiffy.scala [hunk 2]
```diff
@@ -144,6 +143,8 @@ class BigDiffy[T : Coder](@transient val lhs: SCollection[T], @transient val rhs
   private lazy val globalAndFieldStats: SCollection[(GlobalStats, Iterable[FieldStats])] =
     BigDiffy.computeGlobalAndFieldStats(_deltas, ignoreNan)
 
+
+  implicit val deltasCoder: Coder[(MultiKey, String, Any, Any)] = Coder.kryo
```
Comment (idreeskhan) [inline ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/BigDiffy.scala:147]: We should add a comment here about using kryo. This might have to change in a future version

File: ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/BigDiffy.scala [hunk 3]
```diff
@@ -311,7 +312,7 @@ object BigDiffy extends Command with Serializable {
   /** Merge two BigQuery TableSchemas. */
   def mergeTableSchema(x: TableSchema, y: TableSchema): TableSchema = {
     val r = new TableSchema
-    r.setFields(mergeFields(x.getFields.asScala, y.getFields.asScala).asJava)
+    r.setFields(mergeFields(x.getFields.asScala.toList, y.getFields.asScala.toList).asJava)
```
Comment (idreeskhan) [inline ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/BigDiffy.scala:315]: Why this change? Iterator laziness?

File: ratatool-sampling/src/test/scala/com/spotify/ratatool/io/FixRandomData.scala [hunk 4]
```diff
@@ -20,7 +20,7 @@ package com.spotify.ratatool.io
 import com.spotify.ratatool.avro.specific.TestRecord
 
 import java.util.{HashMap => JHashMap, Map => JMap}
-import scala.collection.JavaConversions._
+import scala.jdk.CollectionConverters._
```
Comment (idreeskhan) [inline ratatool-sampling/src/test/scala/com/spotify/ratatool/io/FixRandomData.scala:23]: Should we migrate all instance of `scala.collection.JavaConverters` to this as well? That's the deprecated one now right?

File: ratatool-shapeless/src/main/scala/com/spotify/ratatool/shapeless/CaseClassDiffy.scala [hunk 5]
```diff
@@ -43,7 +43,7 @@ object MapEncoder {
   MapEncoder[FieldType[K, String]] = {
     val name = witness.value.name
     createEncoder { v =>
-      Map(name -> v.self)
+      Map(name -> v.toString)
```
Comment (idreeskhan) [inline ratatool-shapeless/src/main/scala/com/spotify/ratatool/shapeless/CaseClassDiffy.scala:46]: What was the issue here?

Comment (jackdingilian) [reply ratatool-shapeless/src/main/scala/com/spotify/ratatool/shapeless/CaseClassDiffy.scala:46]: In 212 this is a `WrappedString` (https://github.com/scala/scala/blob/v2.12.10/src/library/scala/collection/immutable/WrappedString.scala#L34) 

In 2.13 it is just a `String` so `.self` won't compile.

`toString` in 2.12 returns self anyways so this is equivalent in 2.12 and doesn't break in 2.13.

File: ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/BigDiffy.scala (hunk 3 above)
Comment (jackdingilian) [reply ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/BigDiffy.scala:315]: This expects a `Seq` which in 2.13 is an `scala.collection.immutable.Seq`.  `asScala` returns a `mutable.Buffer[TableFieldSchema]` which extends `mutable.Seq`.  It worked in 2.12 because the default Seq was `scala.collection.Seq` which they both inherit from.  Using toList instead of toSeq so it isn't seen as a redundant method call in 2.12
